packages/backend/src/auth/middleware.ts

```diff
@@ -14,7 +18,22 @@ export function authenticate(
   const { token } = req.cookies
   if (token) {
     const secretKey = config.auth.jwtSecretKey
-    const verification = jwt.verify(token, secretKey)
+    let verification
+
+    try {
+      verification = jwt.verify(token, secretKey)
+    } catch (error) {
+      if (error instanceof TokenExpiredError) {
+        res.status(403).json({ error: 'Your token has expired.' })
+      } else if (error instanceof NotBeforeError) {
+        res.status(403).json({ error: 'Current time is before the nbf claim.' })
+      } else if (error instanceof JsonWebTokenError) {
+        res.status(403).json({ error: 'Invalid signature.' })
+      } else {
+        res.status(403).json({ error: 'Unknown token error.' })
+      }
+    }
```

Comment:
Isn't the code below executed in any case? We should handled that in a better way, I guess

Reply:
Yes, you are right